server/main_server/src/main.cpp

```diff
@@ -1567,6 +1547,128 @@ Environment Variables:
         };
     } // remove_leftover_agent_info_files_for_ips
 
+    auto init_access_time_queue() -> std::optional<std::string>
+    {
+        try {
+            //
+            // Retrieve access time configuration from the catalog.
+            //
+
+            std::optional<std::string> queue_name_prefix;
+            std::optional<std::string> queue_size;
+            std::optional<std::string> resolution_in_seconds;
+
+            const auto role = irods::get_server_property<std::string>(irods::KW_CFG_CATALOG_SERVICE_ROLE);
+
+            // Initialization of the access time queue happens before the agent factory is available.
+            // This means we cannot rely on a local iRODS connection to retrieve the access time configuration 
+            // from the database.
+
+            if (role == irods::KW_CFG_SERVICE_ROLE_PROVIDER) {
+                // Catalog Service Providers are required to have database credentials, so all we have to do
+                // to get around the lack of an agent factory is use nanodbc to interact with the database
+                // directly.
+
+                auto [db_instance, db_conn] = irods::experimental::catalog::new_database_connection();
+                auto row = execute(db_conn, "select option_name, option_value from R_GRID_CONFIGURATION where namespace = 'access_time'");
+
+                while (row.next()) {
+                    const auto opt_name = row.get<std::string>(0);
+
+                    if (opt_name == irods::KW_CFG_ACCESS_TIME_QUEUE_NAME_PREFIX) {
+                        queue_name_prefix = row.get<std::string>(1);
+                    }
+                    else if (opt_name == irods::KW_CFG_ACCESS_TIME_QUEUE_SIZE) {
+                        queue_size = row.get<std::string>(1);
+                    }
+                    else if (opt_name == irods::KW_CFG_ACCESS_TIME_RESOLUTION_IN_SECONDS) {
+                        resolution_in_seconds = row.get<std::string>(1);
+                    }
+                }
+            }
+            else if (role == irods::KW_CFG_SERVICE_ROLE_CONSUMER) {
+                // Catalog Service Consumers aren't guaranteed to have database credentials. We work around
+                // the lack of an agent factory and database credentials by connecting to the Catalog Service
+                // Provider directly and invoking the grid configuration API. This works because consumer
+                // servers are designed to wait for the provider to become available before continuing with
+                // server initialization.
+
+                rodsEnv env;
+                _getRodsEnv(env);
+
+                const auto config_handle = irods::server_properties::instance().map();
+                const auto& config = config_handle.get_json();
+
+                const nlohmann::json::json_pointer json_path{"/catalog_provider_hosts/0"};
```

Comment:
is this assuming the first in the array?    fine for now, just confirming i'm reading the syntax correctly.

Reply:
Yes. We have an issue for that. See #8028.

Reply:
indeed.   will need to remember to look here when that work gets done.   resolving.